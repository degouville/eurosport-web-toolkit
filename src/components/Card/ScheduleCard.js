import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import { rgba } from 'polished';
import * as breakpoints from '../../breakpoints';
import * as colors from '../../colors';

import CardDetails from './CardDetails';

const StyledWrapper = styled.div`
  position: relative;
  height: 169px;
  display: flex;
  border-radius: 2px;
  overflow: hidden;
  box-shadow: 0 2px 3px 0 ${rgba(colors.ebony, 0.3)};

  ${breakpoints.small(css`
    height: 149px;
  `)};
  ${breakpoints.large(css`
    height: 160px;
  `)};
`;

const StyledImage = styled.div`
  background-image: ${({ img }) => `url(${img})`};
  height: 100%;
  width: 100%;
  background-size: cover;
  background-position: 50% 0;
  display: block;
  position: absolute;
  opacity: 0.35;

  :before {
    content: '';
    position: absolute;
    z-index: 1;
    height: 100%;
    width: 101%;
    background: linear-gradient(
      180deg,
      rgba(21, 23, 38, 0) 0%,
      rgba(21, 23, 38, 0.3) 18.66%,
      rgba(21, 23, 38, 0.85) 100%
    );
    opacity: 0.5;
  }
`;

const StyledCardDetails = styled(CardDetails)`
  padding: 20px 20px 10px;
`;

const ScheduleCard = ({ card, ...props }) => (
  <StyledWrapper {...props}>
    <StyledImage img={card.img} />
    <StyledCardDetails card={card} />
  </StyledWrapper>
);

ScheduleCard.propTypes = {
  card: PropTypes.shape({
    img: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    timestamp: PropTypes.string,
    channel: PropTypes.string,
  }),
};

ScheduleCard.defaultProps = {
  card: {
    description: '',
    timestamp: '',
    channel: '',
  },
};

export default ScheduleCard;
