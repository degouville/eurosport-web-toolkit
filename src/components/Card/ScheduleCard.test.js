import React from 'react';
import { shallow } from 'enzyme';
import ScheduleCard from './ScheduleCard';

it('renders a ScheduleCard', () => {
  expect(shallow(<ScheduleCard />)).toMatchSnapshot();
});
