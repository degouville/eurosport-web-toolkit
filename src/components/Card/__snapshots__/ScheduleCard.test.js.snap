// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule card Renders a schedule card 1`] = `
<Styled(div)>
  <Styled(div)
    img="https://i.eurosport.com/2018/10/29/2450727-50913270-2560-1440.jpg?w=200"
  />
  <Styled(Cards.Content)
    card={
      Object {
        "category": "Tennis",
        "channel": "E1",
        "description": "Description",
        "img": "https://i.eurosport.com/2018/10/29/2450727-50913270-2560-1440.jpg?w=200",
        "timestamp": "09:00 - 10:30",
        "title": "Youth olympic summer games",
      }
    }
  />
</Styled(div)>
`;
